### PR TITLE
Update user info route

### PR DESCRIPTION
**Pull request by AdhamHaisami**
Relates #35

----
## Route Description

* udate-user route
* full-route ` /users/:id ` 
* method : `PUT ` 
---- 
## Validation schema

---- 
 ## Request body 
 ```js 
  req.body =  {
       email: adham@adham.adham,
       username: ' ',
       mobile: ' ',
       addtess: ' ',

   }
```
----
#### Success

```js
{ statusCode: 200  ,message: "Login Successfull"
}
```
#### Failed
```js
{statusCode: '400' , message:"user is not authorized,"}
```
----
##  Testing
 * true

### CODE REVIEW

File: server/middleware/validation/updateUser.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      address: string(),
+    });
+    await schema.validate(
+      {
+        username,
+        email,
+        mobile,
+        address,
+      },
+      { abortEalry: false }
```
Comment (okjake) [inline server/middleware/validation/updateUser.js:21]: Typo here, `abortEarly`

File: server/middleware/validation/updateUser.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const { object, string, ref } = requir('yub');
```
Comment (okjake) [inline server/middleware/validation/updateUser.js:1]: `require('yup');`

File: server/router/user.js [hunk 3]
```diff
@@ -3,5 +3,6 @@ const router = require('express').Router();
 const { logoutController } = require('../controller/routes/user');
 
 router.get('/logout', logoutController);
+router.patch('/users');
```
Comment (okjake) [inline server/router/user.js:6]: Where does the user ID come from?

Comment (AdhamHaisami) [reply server/router/user.js:6]: I take it from token. Should I pass it in params?

Comment (okjake) [reply server/router/user.js]: Yeah that would be more REST-ish. It will work either way, but passing it in the params would be clearer

**Review by mariamisa — APPROVED** [review]

**Pull request merged**